sky/clouds/fluidstack.py

```diff
@@ -0,0 +1,337 @@
+                memory=resources.memory,
+                disk_tier=resources.disk_tier)
+            if default_instance_type is None:
+                return ([], [])
+            else:
+                return (_make([default_instance_type]), [])
+
+        assert len(accelerators) == 1, resources
+        acc, acc_count = list(accelerators.items())[0]
+        (instance_list, fuzzy_candidate_list
+        ) = service_catalog.get_instance_type_for_accelerator(
+            acc,
+            acc_count,
+            use_spot=resources.use_spot,
+            cpus=resources.cpus,
+            memory=resources.memory,
+            region=resources.region,
+            zone=resources.zone,
+            clouds='fluidstack')
+        if instance_list is None:
+            return ([], fuzzy_candidate_list)
+        return (_make(instance_list), fuzzy_candidate_list)
+
+    @classmethod
+    def check_credentials(cls) -> Tuple[bool, Optional[str]]:
+
+        try:
+            assert os.path.exists(
+                os.path.expanduser(fluidstack_utils.FLUIDSTACK_API_KEY_PATH))
+            assert os.path.exists(
+                os.path.expanduser(fluidstack_utils.FLUIDSTACK_API_TOKEN_PATH))
+        except AssertionError:
+            return False, (
+                'Failed to access FluidStack Cloud'
+                ' with credentials. '
+                'To configure credentials, go to:\n    '
+                '  https://console.fluidstack.io \n    '
+                'to obtain an API key and API Token, '
+                'then add save the contents '
+                'to ~/.fluidstack/api_key and ~/.fluidstack/api_token \n')
+        except requests.exceptions.ConnectionError:
+            return False, ('Failed to verify FluidStack Cloud credentials. '
+                           'Check your network connection '
+                           'and try again.')
+        return True, None
+
+    def get_credential_file_mounts(self) -> Dict[str, str]:
+        return {filename: filename for filename in _CREDENTIAL_FILES}
+
+    @classmethod
+    def get_current_user_identity(cls) -> Optional[List[str]]:
+        # TODO(mjibril): Implement get_current_user_identity for Fluidstack
+        return None
+
+    def instance_type_exists(self, instance_type: str) -> bool:
+        return service_catalog.instance_type_exists(instance_type, 'fluidstack')
+
+    def validate_region_zone(self, region: Optional[str], zone: Optional[str]):
+        return service_catalog.validate_region_zone(region,
+                                                    zone,
+                                                    clouds='fluidstack')
+
+    @classmethod
+    def check_disk_tier_enabled(cls, instance_type: Optional[str],
+                                disk_tier: DiskTier) -> None:
+        raise exceptions.NotSupportedError(
+            'FluidStack does not support disk tiers.')
+
+    @classmethod
+    def default_username(cls, region: str) -> str:
+        return {
```

Comment:
Is it possible to move this to `images.csv` in the catalog, i.e., we have another column indicate the default username for the image used in each reigon.